pkg/storer/internal/reserve/reserve.go

```diff
@@ -280,57 +281,58 @@ func (r *Reserve) IterateChunksItems(store internal.Storage, startBin uint8, cb
 	return err
 }
 
-func (r *Reserve) EvictBatchBin(ctx context.Context, batch internal.BatchOperation, bin uint8, batchID []byte, cb func(swarm.Chunk)) (int, error) {
-
-	var evicted []*batchRadiusItem
-
-	err := batch.Do(ctx, func(store internal.Storage) error {
-		return store.IndexStore().Iterate(storage.Query{
-			Factory: func() storage.Item { return &batchRadiusItem{} },
-			Prefix:  batchBinToString(bin, batchID),
-		}, func(res storage.Result) (bool, error) {
-			batchRadius := res.Entry.(*batchRadiusItem)
-			evicted = append(evicted, batchRadius)
-			return false, nil
-		})
-	})
-	if err != nil {
-		return 0, err
-	}
-
-	batchCnt := 1000
+// IterateBatchBin iterates over all chunks pertaining to the batch in the particular bin.
+// The callback function should return true to stop the iteration. The stamp on the
+// chunk is not populated as it is not required. This function is used for eviction
+// purposes.
+func (r *Reserve) IterateBatchBin(
+	ctx context.Context,
+	store internal.Storage,
+	bin uint8,
+	batchID []byte,
+	cb func(swarm.Chunk) (bool, error),
+) error {
+	return store.IndexStore().Iterate(storage.Query{
+		Factory: func() storage.Item { return &batchRadiusItem{} },
+		Prefix:  batchBinToString(bin, batchID),
+	}, func(res storage.Result) (bool, error) {
+		batchRadius := res.Entry.(*batchRadiusItem)
 
-	for i := 0; i < len(evicted); i += batchCnt {
-		end := i + batchCnt
-		if end > len(evicted) {
-			end = len(evicted)
+		ch, err := store.ChunkStore().Get(ctx, batchRadius.Address)
+		if err != nil {
+			if errors.Is(err, storage.ErrNotFound) {
+				return false, nil
+			}
+			return true, err
 		}
-
-		err := batch.Do(ctx, func(store internal.Storage) error {
-			for _, item := range evicted[i:end] {
-				c, err := store.ChunkStore().Get(ctx, item.Address)
-				if err != nil {
-					if errors.Is(err, storage.ErrNotFound) {
-						continue
-					}
-					return err
-				}
-
-				cb(c)
-
-				err = removeChunk(ctx, store, item)
-				if err != nil {
-					return err
-				}
+		stamp, err := chunkstamp.LoadWithBatchID(store.IndexStore(), reserveNamespace, batchRadius.Address, batchRadius.BatchID)
```

Comment:
the stamp does not have have to be loaded actually
the batchID is already known, 
the callback can simply return address and batchID